cypress/e2e/read-receipts/read-receipts.spec.ts

```diff
@@ -0,0 +1,133 @@
+         */
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, userName)
+                .then(() => {
+                    cy.createRoom({ name: selectedRoomName }).then((createdRoomId) => {
+                        selectedRoomId = createdRoomId;
+                    });
+                })
+                .then(() => {
+                    cy.createRoom({ name: otherRoomName }).then((createdRoomId) => {
+                        otherRoomId = createdRoomId;
+                    });
+                })
+                .then(() => {
+                    cy.getBot(homeserver, { displayName: botName }).then((botClient) => {
+                        bot = botClient;
+                    });
+                })
+                .then(() => {
+                    // Invite the bot to Other room
+                    cy.inviteUser(otherRoomId, bot.getUserId());
+                    cy.visit("/#/room/" + otherRoomId);
+                    cy.findByText(botName + " joined the room").should("exist");
+
+                    // Then go into Selected room
+                    cy.visit("/#/room/" + selectedRoomId);
+                });
+        });
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it(
+        "Considers room read if there's a receipt for main even if an earlier unthreaded receipt exists #24629",
+        {
+            // When #24629 exists, the test fails the first time but passes later, so we disable retries
+            // to be sure we are going to fail if the bug comes back.
+            // Why does it pass the second time? I wish I knew. (andyb)
+            retries: 0,
+        },
+        () => {
+            // Details are in https://github.com/vector-im/element-web/issues/24629
+            // This proves we've fixed one of the "stuck unreads" issues.
+
+            // Given we sent 3 events on the main thread
+            botSendMessage();
+            botSendMessage().then((main2) => {
+                botSendMessage().then((main3) => {
+                    // (So the room starts off unread)
+                    cy.findByLabelText(`${otherRoomName} 3 unread messages.`).should("exist");
+
+                    // When we send a threaded receipt for the last event in main
+                    // And an unthreaded receipt for an earlier event
+                    cy.sendReadReceipt(fakeEventFromSent(main3));
+                    cy.sendReadReceipt(fakeEventFromSent(main2), "m.read" as any as ReceiptType, true);
+
+                    // (So the room has no unreads)
+                    cy.findByLabelText(`${otherRoomName}`).should("exist");
+
+                    // And we persuade the app to persist its state to indexeddb by reloading and waiting
+                    cy.reload();
+                    cy.findByLabelText(`${selectedRoomName}`).should("exist");
```

Comment:
How does that piece of code persuade the accumulator to persist its state?
Is there a condition somewhere that flushes data to idb before Element is closed?

Reply:
It seems to flush immediately _after_ reload. I don't know why :shrug: